artiq/coredevice/comm_kernel.py

```diff
@@ -38,6 +38,8 @@ class Reply(Enum):
     RPCRequest = 10
 
     ClockFailure = 15
+    # TODO: change this cardinal to fit in Kernel category?
+    KernelAsyncErrorFound = 16
```

Comment:
Just add a flag to the other kernel termination return modes (KernelFinished/KernelException).